src/git.js

```diff
@@ -463,19 +463,19 @@
       var deleted = [];
       var modified = [];
 
-      while (line = lines.shift()) {
-         line = line.split(" ");
-         var st = line.shift();
+      for (lineItem in lines) {
```

Comment:
This introduces an undeclared variable and would iterate over all properties of the lines array rather than just the items contained in it - meaning any enumerable extension to the Array prototype would also be included.